coding/src/reed_solomon/mod.rs

```diff
@@ -80,13 +80,10 @@ impl<H: Hasher> Write for Chunk<H> {
 
 impl<H: Hasher> Read for Chunk<H> {
     /// The maximum size of the shard.
-    type Cfg = crate::Cfg;
+    type Cfg = crate::CodecConfig;
```

Comment:
nit - can we just have this be a type alias for `usize` and document that it's an upper bound, rather than having it be a struct?

If you prefer this for readability, not opposed, though.